cranelift-codegen/meta/src/cdsl/type_inference.rs

```diff
@@ -161,8 +161,8 @@ impl TypeEnvironment {
 
         // Check extra conditions for InTypeset constraints.
         if let Constraint::InTypeset(tv, _) = &constraint {
-            assert!(tv.base.is_none());
-            assert!(tv.name.starts_with("typeof_"));
+            assert!(tv.base.is_none(), "tv.base == {:?}", tv.base);
+            assert!(tv.name.starts_with("typeof_"), "tv.name == {}", tv.name);
```

Comment:
Ditto here: `Name "{}" should start with "type_of"`?

Reply:
```suggestion
            assert!(tv.name.starts_with("typeof_"), "Name \"{}\" should start with \"typeof_\"", tv.name);
```